src/containers/profile/timeline.svelte

```diff
@@ -1,49 +1,96 @@
 <script>
   import { createEventDispatcher } from 'svelte';
-  import Button from 'ui/button.svelte';
+  import { Button, Loading } from 'attractions';
+  import EmptyState from '../../components/common/empty-state.svelte';
   import TimelineEntry from '@/components/profile/timeline-entry.svelte';
+  import LeaveFeedbackModal from '@/components/projects/view/leave-feedback-modal.svelte';
+  import * as api from '@/utils/api.js';
 
   export let timelinePromises;
-
-  const dispatch = createEventDispatcher();
+  export let competences;
+  export let account;
 
   function requestMore() {
     dispatch('more-timeline');
   }
+
+  const leaveFeedbackModal = {
+    open: false,
+    activity: null,
+    application: null,
+    payload: null,
+    async show({ detail: timelinePayload }) {
+      try {
+        leaveFeedbackModal.payload = timelinePayload;
+        const project = await api.json(api.get(`/projects/${timelinePayload.project_id}`));
+        leaveFeedbackModal.activity = project.activities.find(
+          act => act.id === timelinePayload.activity_id,
+        );
+        leaveFeedbackModal.application = leaveFeedbackModal.activity.applications.find(
+          apl => apl.id === timelinePayload.application_id,
+        );
+        leaveFeedbackModal.open = true;
+      } catch (e) {
+        console.error(e);
+      }
+    },
+    async submitFeedback({ detail }) {
+      try {
+        const { value, activity, application } = detail;
+        value.answers = value.answers.map(answer => answer || '');
+        application.feedback = await api.json(api.post(
+          `/projects/${activity.project}/activities/${activity.id}`
+          + `/applications/${application.id}/feedback`,
+          { data: value, csrfToken: account.csrf_token },
+        ));
+        leaveFeedbackModal.payload.feedback_id = application.id;
+        timelinePromises = timelinePromises;
+        account.balance += leaveFeedbackModal.payload.reward;
+        leaveFeedbackModal.open = false;
+      } catch (e) {
+        console.error(e);
+      }
+    },
+  };
+
+  const dispatch = createEventDispatcher();
 </script>
 
 <div class="timeline">
-  {#if timelinePromises.length}
+  {#if timelinePromises}
     {#each timelinePromises as promise}
       {#await promise.then(value => value.data)}
```

Comment:
What is the use of the `.then(value => value.data)` part if the data is not being used?

Reply:
It is used, it's building the timeline, doesn't it?

Reply:
right. I didn't notice the `{:then timelineFragment}` part because the syntax in my mind was just `{#await promise as value}`